sklearn/metrics/pairwise.py

```diff
@@ -791,7 +791,7 @@ def rbf_kernel(X, Y=None, gamma=None):
     return K
 
 
-def cosine_similarity(X, Y=None):
+def cosine_similarity(X, Y=None, dense_output=True):
     """Compute cosine similarity between samples in X and Y.
```

Comment:
While you're editing this function, could you update the docstring to explain what happens if `Y=None`? It seems to not be documented at the moment. I'd add it both in the description as well as under the `Y` param.